suse_migration_services/migration_config.py

```diff
@@ -96,6 +97,16 @@ def update_migration_config_file(self):
                     self.migration_config_file, self.migration_custom_file
                 ).lstrip()
             )
+        else:  # auto-detect baseproduct
+            baseproduct_xml_file = self._read_baseproduct()
+            baseproduct_names = baseproduct_xml_file.findall('name')
+            for name in baseproduct_names:
+                default_migration_target = self.config_data.get('migration_product').split('/')
+                migration_product = '/'.join(
+                    [name.text, default_migration_target[0], default_migration_target[1]]
+                )
+                self.config_data['migration_product'] = migration_product
+                self._write_config_file()
```

Comment:
So this code reads the file behind the baseproduct link, reads in the name tag from the XML and adapts the value we have set in our image for the migration_product that gets passed to the migration plugin for zypper. All of this is only done if no custom migration config file exists. The place for this code is very very misleading. The scope of the MigrationConfig config class and its single responsibility is to handle the migration config file nothing more. This is one of the core rules when writing object oriented and I think we should follow this because it makes sense.

So what you need is code to read the name tag from the base product. If you take a look at units/product_setup.py

you will see that we already have code to handle .prod files. We even have code to detect the base product file. My suggestion is to close this PR and come up with a refactoring that moves any code that handles .prod files into its own class. e.g 

```python
class SUSEProduct(object):
    ...
```

Move all code we already have to the new class and just add the little extension to read the name tag
This should be imho the first PR. Let's review that one and then continue